lib/std/math/big/int.zig

```diff
@@ -58,7 +82,16 @@ pub fn calcSetStringLimbsBufferLen(base: u8, string_len: usize) usize {
 }
 
 pub fn calcSetStringLimbCount(base: u8, string_len: usize) usize {
-    return (string_len + (limb_bits / base - 1)) / (limb_bits / base);
+    const logs2 = comptime blk: {
+        var ar = [_]f32{0} ** 37;
+        for (2..37) |base_i| {
+            ar[base_i] = math.log2(@intToFloat(f32, base_i));
+        }
+        break :blk ar;
+    }; // size of a digit of each base in bits
+    const bits = @floatToInt(usize, logs2[base] * @intToFloat(f32, string_len)) + 1;
```

Comment:
Please add the explicit limit for which inputs are accurate for using f32. See https://blog.demofox.org/2017/11/21/floating-point-precision/ "Storing Integers":
"In fact, a floating point number can EXACTLY store all integers from -2^{MantissaBits+1} to +2^{MantissaBits+1}.
For floats, it’s -16,777,216 to +16,777,216. (\pm 2^{24})
For doubles it’s -9,007,199,254,740,992 to +9,007,199,254,740,992. (\pm 2^{53})"

Reply:
Should I use an `assert` to ensure the product will always be stored accurately ? The limit for which it is accurate depend on the base, for instance: using `f32`, the limit for `string_len` is 3245156 at base 36 but 5050445 at base 10.

Reply:
I think that would be a good idea as this is only 1 multiplication and 1 comparison to ensure correctness for the floats instead of accidentally broken code and all of those can be prefetched.